ui/components/app/flask/snap-install-warning/snap-install-warning.js

```diff
@@ -1,20 +1,41 @@
-import React, { useCallback, useState } from 'react';
+import React, { useCallback, useReducer } from 'react';
 import PropTypes from 'prop-types';
+import { produce } from 'immer';
 import { useI18nContext } from '../../../../hooks/useI18nContext';
 import CheckBox from '../../../ui/check-box/check-box.component';
 import Typography from '../../../ui/typography/typography';
 import { TYPOGRAPHY } from '../../../../helpers/constants/design-system';
 import Popover from '../../../ui/popover';
 import Button from '../../../ui/button';
 
-export default function SnapInstallWarning({ onCancel, onSubmit, snapName }) {
+/**
+ * a very simple reducer using produce from Immer to keep checkboxes state manipulation
+ * immutable and painless.
+ */
+const checkboxStateReducer = produce((state, action) => {
+  switch (action.type) {
+    case 'check':
+      state[action.checkboxId] = state[action.checkboxId]
```

Comment:
Not sure if cleaner, but `!state[action.checkboxId]` should be enough here?

Reply:
state[action.checkboxId] can be undefined, I don't think it will work in this case

Reply:
`!undefined === true`
`!null === true`